src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -776,9 +779,9 @@ CmdletProviderContext normalizePathContext
         #region push-Pop current working directory
 
         /// <summary>
-        /// A bounded stack for the location history of Set-Location
+        /// Location history for Set-Location that supports Undo/Redo using bounded stacks.
         /// </summary>
-        private BoundedStack<PathInfo> _SetLocationHistory;
+        private HistoryStack<PathInfo> _SetLocationHistory;
```

Comment:
Please use _setLocationHistory for private.
And seems it is `private readonly`.